### PR TITLE
[FEAT] : Note color 🎨 and Pin 📌 functionality added

### PR DESCRIPTION
**Pull request by t007rushi**
Note color 🎨 
- > You can color notes while creation 
- >  Can update already created ones

and Pin 📌 
- > Pin the note while creation 
- > can pin/unpin after created

recording

<details><summary>Browser metadata</summary>

```
Path:      /notes
Browser:   Chrome 99.0.4844.84 on Windows 10
Viewport:  1920 x 969 @1x
Language:  en-US
Cookies:   Enabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-8--monkeep.netlify.app%2Fnotes&os=Windows&os_version=10&browser=Chrome&browser_version=99.0&resolution=1920x969&speed=1&start=true&ref=netlify-source)

</details>

[Open Deploy Preview](https://deploy-preview-8--monkeep.netlify.app/notes) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=62480b6dc59153000b22fe66&commentID=62480c76db897524c1152439&resolution=resolved)

### CODE REVIEW

File: src/components/CreateNote/CreateNote.jsx [hunk 1]
```diff
@@ -1,81 +1,81 @@
...
 export const CreateNote = () => {
   const [expand, setExpand] = useState(false);
+
+  const [description, setDescription] = useState("");
   const closerEditor = () => {
     setExpand(false);
   };
   const expandEditor = () => {
     setExpand(true);
   };
```
Comment (JantuDeb) [inline src/components/CreateNote/CreateNote.jsx:19]:
```
onst closerEditor = () =>setExpand(false);
  
  const expandEditor = () =>setExpand(true);
  

```

File: src/components/CreateNote/CreateNote.jsx [hunk 2]
```diff
@@ -1,81 +1,81 @@
...
+          <Editor handleInput={handleInput} description={description} />
           <nav className="flex-row note-options">
             <input
               type="text"
               placeholder="enter label"
               className="label-txt"
-              onChange={(e) =>
-                setNotedata({ ...notedata, label: e.target.value })
-              }
+              onChange={(e) => {}}
```
Comment (JantuDeb) [inline src/components/CreateNote/CreateNote.jsx:64]: ??

**Review by JantuDeb — APPROVED** [review]
This is really awesome  UI you made.

File: src/components/CreateNote/CreateNote.jsx (hunk 2 above)
Comment (t007rushi) [reply src/components/CreateNote/CreateNote.jsx:64]: future ref for tags

File: src/components/Editor/Editor.jsx [hunk 3]
```diff
@@ -2,10 +2,11 @@ import ReactQuill from "react-quill";
 import "react-quill/dist/quill.snow.css";
 import "./editor.css";
 
-export default function Editor({ handleInput }) {
+export default function Editor({ handleInput, description }) {
```
Comment (Priya-Patel31) [inline src/components/Editor/Editor.jsx]: avoid default imports .

File: src/components/Note/Note.jsx [hunk 4]
```diff
@@ -1,34 +1,58 @@
 import React from "react";
-import { useNotes } from "../../context/notes-context";
 import "./note.css";
+import { useNotes } from "../../context/notes-context";
 import { FiTrash2 } from "react-icons/fi";
 import { RiInboxArchiveLine } from "react-icons/ri";
-import { MdOutlineColorLens } from "react-icons/md";
+import { BsFillPinFill, BsPin } from "react-icons/bs";
```
Comment (Priya-Patel31) [inline src/components/Note/Note.jsx:6]: Suggestion : You can have icons.js file and then import these icons from there.

File: src/constants/notesConstant.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export const PIN ="PIN"
```
Comment (Priya-Patel31) [inline src/constants/notesConstant.js:1]: Really liked the way you have made separate  file for constants.

**Review by Priya-Patel31 — APPROVED** [review]
The code is well structured and very well formatted . Liked the way you have separate files for constants . You can go ahead merge it .

File: src/components/Note/Note.jsx (hunk 4 above)
Comment (t007rushi) [reply src/components/Note/Note.jsx:6]: nice idea

**Pull request merged**